libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -942,20 +942,23 @@ string YulUtilFunctions::accessCalldataTailFunction(Type const& _type)
 		return Whiskers(R"(
 			function <functionName>(base_ref, ptr_to_tail) -> addr<?dynamicallySized>, length</dynamicallySized> {
 				let rel_offset_of_tail := calldataload(ptr_to_tail)
-				if iszero(slt(rel_offset_of_tail, sub(sub(calldatasize(), base_ref), sub(<neededLength>, 1)))) { revert(0, 0) }
+				if iszero(slt(rel_offset_of_tail, sub(sub(calldatasize(), base_ref), sub(<neededLength>, 1)))) { <invalidCalldataTailOffset> }
 				addr := add(base_ref, rel_offset_of_tail)
 				<?dynamicallySized>
 					length := calldataload(addr)
-					if gt(length, 0xffffffffffffffff) { revert(0, 0) }
-					if sgt(base_ref, sub(calldatasize(), mul(length, <calldataStride>))) { revert(0, 0) }
+					if gt(length, 0xffffffffffffffff) { <invalidCalldataTailLength> }
 					addr := add(addr, 32)
+					if sgt(addr, sub(calldatasize(), mul(length, <calldataStride>))) { <shortCalldataTail> }
```

Comment:
This check was actually incorrect before, causing the ``calldata_array_dynamic_invalid`` and ``calldata_array_dynamic_invalid_static_middle`` tests in SolidityEndToEndTest.cpp to fail. Comparing with the old version should and/or thinking about it for a second should make it clear that this change is needed as fix.

Reply:
Wait, this check was incorrect in develop?

Reply:
For Sol->Yul, yes.

Reply:
The one for regular/old codegen was correct - and the sol->yul part was supposed to do exactly the same, but this particular check was transferred incorrectly (actually my bad ;-)). Since this PR makes the old and the new code generators use the exact same code for this, more tests were run on it and I noticed. Since it's not the worst of bugs and sol->yul is experimental I wouldn't say it's a big deal, though. But for reviewing it would be nice to recheck that the yul function does exactly the same as the two inline assembly snippets this PR now removes.

Reply:
Ok, sounds good!